src/agent/src/image_rpc.rs

```diff
@@ -47,6 +77,227 @@ impl ImageService {
         }
     }
 
+    fn build_oci_path(cid: &str) -> PathBuf {
+        let mut oci_path = PathBuf::from("/tmp");
+        oci_path.push(cid);
+        oci_path.push(IMAGE_OCI);
+        oci_path
+    }
+
+    fn decrypt_layer_data(
+        layer: &ImageLayer,
+        layer_digest: &str,
+        image_manifest: &mut OciManifest,
+        crypto_config: &CryptoConfig,
+        oci_blob_path: &Path,
+    ) -> Result<()> {
+        if let Some(decrypt_config) = &crypto_config.decrypt_config {
```

Comment:
This function is only called for encrypted layers. But if the function returns `Ok(())` if there is no decrypt config, what happens next? Shouldn't this function error out if there is no decrypt config since the layer will surely be unusable unless you can decrypt it...?

Reply:
Return `Err` when no decrypt config avaiable